elfi/examples/scratch_assay.py

```diff
@@ -0,0 +1,192 @@
+            if cell_arr[coords[0], coords[1]] == 0:
+                # move
+                cell_arr[cell_coords[cell][0], cell_coords[cell][1]] = 0
+                cell_arr[coords[0], coords[1]] = 1
+                # update coordinate list
+                cell_coords[cell] = coords
+
+        # proliferation events
+
+        candidates = random_state.choice(num_cells, size=num_cells)
+        p = random_state.uniform(size=num_cells)
+        candidates = candidates[p < pp]
+
+        for cell in candidates:
+            coords = _random_move(cell_coords[cell], nrows, ncols, random_state)
+            cell_arr[coords[0], coords[1]] = 1   # increases cell count if coords is unoccupied
+
+        # observations
+
+        if (iteration + 1) % obs_interval == 0:
+            obs_arr[int((iteration + 1)/obs_interval)] = np.copy(cell_arr)
+
+    return np.transpose(obs_arr, (1, 2, 0))
+
+
+def cell_summaries(x):
+    """Calculate summary statistics proposed by Price et al (2018).
+
+    Parameters
+    ----------
+    x : np.ndarray
+        Simulated/observed data in shape (batch_size, nrows, ncols, num_obs).
+
+    Returns
+    -------
+    np.ndarray (batch_size, num_obs+1)
+
+    """
+    # 1. mismatch between consecutive observations
+    ds = np.sum(np.abs((x[:, :, :, : -1] - x[:, :, :, 1:])), axis=(1, 2))
+
+    # 2. final cell count
+    count = np.sum(x[:, :, :, -1], axis=(1, 2))[:, None]
+
+    return np.concatenate((ds, count), axis=1)
+
+
+def get_model(true_params=None, init_arr=None, init_params=None, seed_obs=None):
+    """
+    Return complete scratch assay model.
+
+    Parameters
+    ----------
+    true_params : list, optional
+        Parameters with which the observed data is generated.
+    init_arr : np.ndarray, optional
+        Initial cell locations as binary array.
+    init_params : list, optional
+        Parameters for random initialisation. Not used if init_arr is provided.
+    seed_obs : int, optional
+        Seed for the observed data generation.
+
+    Returns
+    -------
+    elfi.ElfiModel
+
+    """
+    if true_params is None:
+        true_params = [0.25, 0.002]
+
+    m = elfi.ElfiModel()
+
+    # priors
+    elfi.Prior('uniform', 0, 1, model=m, name='pm')
+    elfi.Prior('uniform', 0, 0.01, model=m, name='pp')
+
+    # observed data
+    random_state = np.random.RandomState(seed_obs)
+    obs = cell_sim(*true_params, init_arr, init_params, random_state=random_state)
+
+    # simulator
+    cell_sim_vector = elfi.tools.vectorize(cell_sim, constants=(2,))
+    init_arr = obs[:, :, 0]
+    obs = obs[None, :]
+    elfi.Simulator(cell_sim_vector, m['pm'], m['pp'], init_arr, name='sim', observed=obs)
+
+    # summaries
+    elfi.Summary(cell_summaries, m['sim'], name='sums')
+
+    # distance
+    num_ds = m['sums'].observed.size - 1
+    num_init = np.sum(init_arr)
+    weis = np.concatenate((np.ones(num_ds)/num_ds, np.array([1])))/num_init**2
+    elfi.Distance('euclidean', m['sums'], w=weis, name='d')
```

Comment:
Would it make sense to wrap this to a single custom discrepancy-node for clarity?